ddd-core/src/test/kotlin/io/paju/ddd/AggregateRootTest.kt

```diff
@@ -23,9 +23,9 @@ internal class AggregateRootTest {
     @Test
     fun reconstruct() {
         val aggregate = makeAggregate()
-        val reconstructed = CounterAggregate(aggregate.id).apply {
-            reconstruct(aggregate.uncommittedChanges())
-        }
+        val reconstructed = AggregateRootBuilder
+            .build{CounterAggregate(aggregate.id)}
+            .fromEvents(aggregate.uncommittedChanges())
         assertEquals(1, reconstructed.state().counter)
         assertEquals(0, reconstructed.uncommittedChanges().size)
```

Comment:
The version update logic is not fully implemented and planned. I will start work with that next and try to figure out how it should be done. Changes for this will come in other PR.

The issue about version handling is this https://github.com/nemit/paju-ddd/issues/25